File: ats-ness-front-prod/src/app/demo/models/societe.ts
import { Besoin } from "./besoin";
import {Client} from "./Client";


export interface Societe {
  id?: number; // optional because it may not be set for new Societe instances
  dateDeCreation?: string; // LocalDateTime; should be a string in Angular
  dateDeDerniereMiseAJour?: string;
  logo?: string;
  nom: string;
  societeMere?: string;
  adresse?: string;
  capitalSocial: number;
  rcs?: string;
  villeRcs?: string;
  siret: number;
  secteur?:string;

    organigramme?: string[]; // Array of URLs to the organigramme files
  besoins?: Besoin[]; // Array of Besoin objects
  clients?: Client[]; // Array of Besoin objects

}

File: ats-ness-back-prod/src/main/java/nesstechnologies/fr/applicationesnbackend/entities/Societe.java
package nesstechnologies.fr.applicationesnbackend.entities;

import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
@ToString(exclude = "clients")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Societe implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private LocalDateTime dateDeCreation;
    private LocalDateTime dateDeDerniereMiseAJour;

    private String logo;
    private String nom;
    private String societeMere;
    private String adresse;
    private float capitalSocial;
    private String rcs;
    private String villeRcs;
    private int siret;
    private String secteur;

    @ElementCollection
    private List<String> organigramme;
    @OneToMany(mappedBy = "societe", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<Besoins> besoins;

    @OneToMany(mappedBy = "societe")
    @JsonManagedReference
    private List<Client> clients;



}